### PR TITLE
fix ldap and "last logged in" handling

### PR DESCRIPTION
**Pull request by adrianschroeter**

### CODE REVIEW

File: src/api/app/models/user.rb [hunk 1]
```diff
@@ -220,7 +220,69 @@ def state=(value)
...
+        logger.debug "#{login} not found in LDAP."
+        ldap_info = nil # now fall through as if we'd not found a user
+      end
+
+      return nil unless ldap_info
+
+      # We've found an ldap authenticated user - find or create an OBS userDB entry.
+      if user
+        # Check for ldap updates
+        if user.email != ldap_info[0] or user.realname != ldap_info[1]
```
Comment (mdeniz) [inline src/api/app/models/user.rb]: Don't use `or` use `||` instead ;)

File: src/api/app/models/user.rb [hunk 2]
```diff
@@ -220,7 +220,69 @@ def state=(value)
...
+          user.email = ldap_info[0]
+          user.realname = ldap_info[1]
+          user.save
+        end
+        user.login_failure_count = user.login_failure_count + 1
```
Comment (mdeniz) [inline src/api/app/models/user.rb]: Maybe I'm missing something... but increasing the login_**failure**_count every time you found the user from ldap is ok? I think that this should reset the failure count not increase it (like at line 228 of this file before you changed it).

File: src/api/app/models/user.rb [hunk 3]
```diff
@@ -373,7 +443,7 @@ def authenticate(user_login, password)
       # User account is not confirmed yet
       return if user.try(:state) == STATES['unconfirmed']
 
-      Rails.logger.debug "Authentificated user '#{user.try(:login)}'"
+      Rails.logger.debug "Authentificated user #{user.inspect} '#{user.try(:login)}'"
```
Comment (mdeniz) [inline src/api/app/models/user.rb]: If you inspect the whole user object, then don't do `user.try(:login)`. Or remove the inspect if this was some debug logging at development.

File: src/api/config/environment.rb [hunk 4]
```diff
@@ -18,13 +18,5 @@
 CONFIG['proxy_auth_mode'] ||= :off
 CONFIG['frontend_ldap_mode'] ||= :off
 
-# use hardcoded source server port for test and development
```
Comment (mdeniz) [inline src/api/config/environment.rb]: Why did you need to remove these lines? Now, what are the port numbers?

File: src/api/app/models/user.rb [hunk 5]
```diff
@@ -220,15 +220,85 @@ def state=(value)
...
+        logger.debug "#{login} not found in LDAP."
+        ldap_info = nil # now fall through as if we'd not found a user
+      end
 
-    # If the user could be found and the passwords equal then return the user
-    if user && user.password_equals?(password)
-      if user.login_failure_count > 0
+      unless ldap_info
+        if user
+          user.login_failure_count = user.login_failure_count + 1
```
Comment (mdeniz) [inline src/api/app/models/user.rb]: Maybe I'm missing something... but increasing the login_failure_count every time you found the user from ldap is ok? I think that this should reset the failure count not increase it (like at line 228 of this file before you changed it).

File: src/api/config/environment.rb (hunk 4 above)
Comment (adrianschroeter) [inline src/api/config/environment.rb]: they are already defined in config/environments/test.rb and developent.rb. I have just removed the double definition.

File: src/api/app/models/user.rb (hunk 3 above)
Comment (adrianschroeter) [inline src/api/app/models/user.rb]: indeed, will revert that

File: src/api/config/environment.rb (hunk 4 above)
Comment (mdeniz) [inline src/api/config/environment.rb]: Perfect then :+1:
